panic with kfake.Superuser wrong method

To reproduce:

```go
package main

import (
	"fmt"
	"os"

	"github.com/twmb/franz-go/pkg/kfake"
)

func main() {
	cluster, err := kfake.NewCluster(
		kfake.EnableSASL(),
		kfake.Superuser("wrong", "user", "pass"),
	)
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	defer cluster.Close()

	fmt.Println(cluster.ListenAddrs())
}
```

Output:

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x158 pc=0x6b9e77]

goroutine 1 [running]:
github.com/twmb/franz-go/pkg/kfake.(*Cluster).Close(0xc0000dfaf0?)
        /home/unikum/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20230905214301-01651affd204/cluster.go:196 +0x17
github.com/twmb/franz-go/pkg/kfake.NewCluster.func1()
        /home/unikum/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20230905214301-01651affd204/cluster.go:112 +0x25
github.com/twmb/franz-go/pkg/kfake.NewCluster({0xc0000dff08, 0x2, 0xc0000061a0?})
        /home/unikum/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20230905214301-01651affd204/cluster.go:134 +0xa39
main.main()
        /mnt/ssd1/work/centrofinans/dev/pg2kafka/tmp/main.go:11 +0xce
exit status 2
```

--- a/pkg/kfake/cluster.go
+++ b/pkg/kfake/cluster.go
@@ -68,7 +68,7 @@ func MustCluster(opts ...Opt) *Cluster {
 }
 
 // NewCluster returns a new mocked Kafka cluster.
-func NewCluster(opts ...Opt) (c *Cluster, err error) {
+func NewCluster(opts ...Opt) (*Cluster, error) {
 	cfg := cfg{
 		nbrokers:        3,
 		logger:          new(nopLogger),
@@ -87,7 +87,7 @@ func NewCluster(opts ...Opt) (c *Cluster, err error) {
 		cfg.nbrokers = len(cfg.ports)
 	}
 
-	c = &Cluster{
+	c := &Cluster{
 		cfg: cfg,
 
 		adminCh:      make(chan func()),
@@ -107,6 +107,7 @@ func NewCluster(opts ...Opt) (c *Cluster, err error) {
 	}
 	c.data.c = c
 	c.groups.c = c
+	var err error
 	defer func() {
 		if err != nil {
 			c.Close()
@@ -147,9 +148,9 @@ func NewCluster(opts ...Opt) (c *Cluster, err error) {
 		if len(cfg.ports) > 0 {
 			port = cfg.ports[i]
 		}
-		ln, err := newListener(port, c.cfg.tls)
+		var ln net.Listener
+		ln, err = newListener(port, c.cfg.tls)
 		if err != nil {
-			c.Close()
 			return nil, err
 		}
 		b := &broker{

